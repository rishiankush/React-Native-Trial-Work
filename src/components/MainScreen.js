/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

 import React, {useState} from 'react';
 import {
   StyleSheet,
   Text,
   View,
 } from 'react-native';
 import { Button, Paragraph, Dialog, Caption } from 'react-native-paper';
 import DateTimePickerModal from "react-native-modal-datetime-picker";
 import moment from 'moment';
 import { StackActions } from '@react-navigation/native';
 
 const MainScreen = ({navigation}) => {
    const [isDatePickerVisible, setDatePickerVisibility] = useState(false);
    const [dialogMessage, setDialogMessage] = useState('')
    const [visible, setVisible] = React.useState(false);

    const hideDialog = () => setVisible(false);

    const showDatePicker = () => {
      setDatePickerVisibility(true);
    };
  
    const hideDatePicker = () => {
      setDatePickerVisibility(false);
    };
  
    const handleConfirm = (date) => {
    //   console.log("A date has been picked: ", date.toISOString());
      let isoDate = date.toISOString()
      let checkWeekDay = moment(isoDate).weekday();
    //   console.log('weekday check ***** ', checkWeekDay)
      if(checkWeekDay > 5 || checkWeekDay === 0){
        setVisible(true)
        setDialogMessage(`Selected day is a weekend`);
      }else{
        navigation.dispatch(
            StackActions.push('OutputScreen',{date: isoDate})
        );
      }
      hideDatePicker();
    };
   return (
    <View style={{flex:1,alignItems:'center'}}>
        <Caption style={{marginTop: '20%'}}>Please Note: You can only select the weekdays or business days.</Caption>
        <Button style={{width:'80%',marginTop: '10%'}} mode="contained" onPress={showDatePicker}>
            Pick a date
        </Button>
        <DateTimePickerModal
            isVisible={isDatePickerVisible}
            mode="date"
            // minimumDate={new Date()}
            onConfirm={handleConfirm} 
            onCancel={hideDatePicker}
        />
        <Dialog visible={visible} onDismiss={hideDialog}>
            <Dialog.Content>
                <Paragraph>{dialogMessage}</Paragraph>
            </Dialog.Content>
        </Dialog>
    </View>
   );
 };
 
 export default MainScreen;
 