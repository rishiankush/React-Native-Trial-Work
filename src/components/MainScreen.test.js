import React from 'react';
import renderer from 'react-test-renderer';
import MainScreen from './MainScreen';

test('renders correctly', () => {
  const tree = renderer.create(<MainScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});