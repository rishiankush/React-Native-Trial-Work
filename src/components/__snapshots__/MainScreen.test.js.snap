// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontFamily": "System",
            "fontWeight": "400",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 12,
              "letterSpacing": 0.4,
              "lineHeight": 20,
              "marginVertical": 2,
            },
            Object {
              "marginTop": "20%",
            },
          ],
        ],
      ]
    }
  >
    Please Note: You can only select the weekdays or business days.
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#6200ee",
        "borderColor": "transparent",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 0,
        "elevation": 2,
        "marginTop": "10%",
        "minWidth": 64,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 0.75,
          "width": 0,
        },
        "shadowOpacity": 0.24,
        "shadowRadius": 1.5,
        "width": "80%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 4,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "letterSpacing": 1,
                  "marginHorizontal": 16,
                  "marginVertical": 9,
                  "textAlign": "center",
                },
                undefined,
                Object {
                  "textTransform": "uppercase",
                },
                Object {
                  "color": "#ffffff",
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                Object {
                  "fontFamily": "System",
                  "fontWeight": "500",
                },
                undefined,
              ],
            ]
          }
        >
          Pick a date
        </Text>
      </View>
    </View>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    isVisible={false}
    onHide={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
  </Modal>
</View>
`;
